elasticdl/python/master/servicer.py

```diff
@@ -229,11 +223,19 @@ def _update_model(self):
 
         self._opt.apply_gradients(grad_var)
 
+        # need the lock for model version update in async SGD
+        if self._use_async:
+            self._lock.acquire()
         self._update_model_version()
-        self._gradient_sum.clear()
-        self._gradient_sum_indexed.clear()
-        self._edl_embedding_gradients.clear()
-        self._grad_n = 0
+        self._update_evaluation()
```

Comment:
Can we wrap these clauses into a `with` statemennt?

Reply:
May not easy to put `with` here, as the lock is only for `_use_async`.